bundles/org.openhab.ui.start/src/main/java/org/openhab/ui/start/internal/RootServlet.java

```diff
@@ -0,0 +1,172 @@
+ *
+ * @author Kai Kreuzer - Initial contribution
+ *
+ */
+@Component(immediate = true)
+public class RootServlet extends HttpServlet {
+
+    private static final long serialVersionUID = -2091860295954594917L;
+
+    private final Logger logger = LoggerFactory.getLogger(RootServlet.class);
+
+    protected HttpService httpService;
+
+    // an enumeration for the state the whole system is in
+    private enum LifeCycleState {
+        STARTING,
+        STARTED,
+        STOPPING,
+        UPDATING;
+    }
+
+    private String page404;
+    private String pageStatus;
+
+    private DashboardReady dashboardStarted;
+    private LifeCycleState lifecycleState = LifeCycleState.STARTING;
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        if (dashboardStarted != null) {
+            // all is up and running
+            if (req.getRequestURI().equals("/")) {
+                resp.sendRedirect("/start/index");
+            } else {
+                resp.sendError(HttpServletResponse.SC_NOT_FOUND);
+                resp.setContentType("text/html;charset=UTF-8");
+                resp.getWriter().append(page404);
+                resp.getWriter().close();
+            }
+        } else {
+            // report current system state
+            String message = null;
+            String subMessage = null;
+            switch (lifecycleState) {
+                case STARTING:
+                    message = "openHAB is starting...";
+                    subMessage = "Please wait a moment!";
+                    break;
+                case UPDATING:
+                    message = "openHAB is updating...";
+                    subMessage = "Please wait a moment!";
+                    break;
+                case STOPPING:
+                    message = "openHAB is shutting down...";
+                    subMessage = "Please stand by.";
+                    break;
+                default:
+                    throw new IllegalStateException("Invalid system state " + lifecycleState);
+            }
+            resp.setContentType("text/html;charset=UTF-8");
+            resp.getWriter().append(pageStatus.replace("${message}", message).replace("${submessage}", subMessage));
+            resp.getWriter().close();
+        }
+    }
+
+    @Activate
+    protected void activate(ComponentContext context) {
+        try {
+            httpService.registerServlet("/", this, new Properties(), httpService.createDefaultHttpContext());
+        } catch (ServletException | NamespaceException e) {
+            logger.error("Failed registering root servlet!", e);
+        }
+        URL notfound = context.getBundleContext().getBundle().getEntry("pages/404.html");
+        if (notfound != null) {
+            try {
+                page404 = IOUtils.toString(notfound.openStream());
+            } catch (IOException e) {
+                throw new RuntimeException(e);
```

Comment:
Do not throw raw exception types such as the `Exception` or the `RuntimeException`, use a specific exception and if none exists create your own.